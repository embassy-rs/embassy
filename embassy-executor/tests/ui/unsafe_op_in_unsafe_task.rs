#![cfg_attr(feature = "nightly", feature(impl_trait_in_assoc_type))]
#![deny(unsafe_op_in_unsafe_fn)]

#[embassy_executor::task]
async unsafe fn task() {
    let x = 5;
    (&x as *const i32).read();
}

fn main() {}
