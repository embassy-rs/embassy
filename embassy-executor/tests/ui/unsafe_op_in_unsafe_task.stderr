error[E0133]: call to unsafe function `std::ptr::const_ptr::<impl *const T>::read` is unsafe and requires unsafe block
 --> tests/ui/unsafe_op_in_unsafe_task.rs:7:5
  |
7 |     (&x as *const i32).read();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
  |
  = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html>
  = note: consult the function's documentation for information on how to avoid undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
 --> tests/ui/unsafe_op_in_unsafe_task.rs:5:1
  |
5 | async unsafe fn task() {
  | ^^^^^^^^^^^^^^^^^^^^^^
note: the lint level is defined here
 --> tests/ui/unsafe_op_in_unsafe_task.rs:2:9
  |
2 | #![deny(unsafe_op_in_unsafe_fn)]
  |         ^^^^^^^^^^^^^^^^^^^^^^
